sdjournal/journal.go

```diff
@@ -260,6 +262,61 @@ func (j *Journal) GetDataValue(field string) (string, error) {
 	return strings.SplitN(val, "=", 2)[1], nil
 }
 
+// GetDataMap returns all the Key-Value pairs of data for the current journal
+// entry. Mimics the behavior of output_export in logs-show.c
+func (j *Journal) GetDataMap() (map[string]string, error) {
+	var r C.int
+	var d unsafe.Pointer
+	var l C.size_t
+
+	resultMap := make(map[string]string)
+
+	j.mu.Lock()
+	defer j.mu.Unlock()
+
+	var realtimeUsec C.uint64_t
```

Comment:
I am not quite sure how I feel about muxing these fields in with the data map - from http://www.freedesktop.org/software/systemd/man/systemd.journal-fields.html#Address%20Fields : 
>  Note that these are not proper fields when stored in the journal but for addressing metadata of entries. 

perhaps we could have something like
```
type JournalEntry {
   fields map[string]string
   Cursor string
   RealtimeTimestamp uint64
   MonotonicTimestamp uint64
}
```
or similar, then instead of GetDataMap, `GetJournalEntry() (*JournalEntry, error)`

@vcaputo any thoughts?